src/Workspaces/Remote/ServiceHub/Shared/ServiceHubServiceBase.cs

```diff
@@ -130,6 +113,15 @@ public virtual void Initialize(PinnedSolutionInfo info)
 
         public void Dispose()
         {
+            if (_disposed)
+            {
+                // guard us from double disposing. this can happen in unit test
+                // due to how we create test mock service hub stream that tied to
+                // remote host service
+                return;
+            }
+
+            _disposed = true;
             Rpc.Dispose();
             _shutdownCancellationSource.Dispose();
 
```

Comment:
:memo: The line below this says `Dispose(false)`, but it should be `Dispose(true)`

Reply:
sure